### PR TITLE
Fix bug in comparison operators 

### LINKED ISSUE #135: Bug in __gte for string filters
**Describe the bug**
When providing a string filter to `__gte` the query fails if the string contains a `-` character

**To Reproduce**
1. Run `soul-cli`, connecting to `chinook.db`
2. Enter this URL: `http://localhost:8000/api/tables/invoices/rows?_filters=BillingAddress__gte:Theodor`
3. Note that there are 21 matching records, and the first has a BillingAddress of `Theodor-Heuss-Straße 34`
4. Now filter for address including the hyphen: `Theodor-Heuss`. So a URL of: `http://localhost:8000/api/tables/invoices/rows?_filters=BillingAddress__gte:Theodor-Heuss`
5. Note that `Soul` returns an error:
> {"message":"near \"LIMIT\": syntax error","error":{"code":"SQLITE_ERROR"}}

**Note:**
We seem to have a related issue with filters on InvoiceDate that contain a "-".
> http://localhost:8000/api/tables/invoices/rows?_filters=InvoiceDate:2009-01-01%2000:00:00

Returns this:
> {"message":"near \".01\": syntax error","error":{"code":"SQLITE_ERROR"}}

**Note 2:**

The invoices are all dated 2009.  So this query reports 412 matching records:
> http://localhost:8000/api/tables/invoices/rows?_page=1&_limit=20&_filters=InvoiceDate__gt:2009

But, if I change the year to 2030, there are still 412 matching records:
> http://localhost:8000/api/tables/invoices/rows?_page=1&_limit=20&_filters=InvoiceDate__gt:2030

In contrast, using a `__lt` operator, records are never returned:
> http://localhost:8000/api/tables/invoices/rows?_page=1&_limit=20&_filters=InvoiceDate__lt:2030

These suggest `__lt`, `__gt`, `__lte` and `__gte` aren't working properly. 

Separately to that, the query parsing is failing if the filter value includes a hyphen, like: `2009-01`.

**Expected behavior**
I expected the 21 matching records to still be returned

### PR DESCRIPTION
**Pull request by AbegaM**
PR Sponsored By @IanMayo 

Fixes #135 

### Modifications 

The comparison operators __gte, __lte, __gt, and __lt were not working for string values with a hyphen. I have changed the regex syntax.

### CODE REVIEW

**Comment by IanMayo** [comment]
Sorry Abenezer, I'm nervous about the changes here, and the risk of this fix breaking something else:
image

Aah, we do have our unit tests, they should verify that existing functionality isn't broken.

Talking of which could you please introduce a test in `rows.test.js` that verifies this is working?

**Comment by AbegaM** [comment]
@IanMayo No problem, let me run the test and i will add additional tests if necessary

**Comment by IanMayo** [comment]
I've checked out your branch locally, but this query still returns invoices dated 2009:
```
http://localhost:8000/api/tables/invoices/rows?_filters=InvoiceDate__gte:2030
```

**Comment by IanMayo** [comment]
The regex was introduced to allow filter ids to be expressed as an array `_filters:invoiceId:[22,33]`.  I confirm this is still working with the new regex.

File: core/src/controllers/rows.test.js [hunk 1]
```diff
@@ -69,6 +69,18 @@ describe('Rows Endpoints', () => {
     expect(res.body.data[0].lastName).not.toBeNull();
   });
 
+  it('GET /tables/:name/rows: should filter items by createdAt date', async () => {
+    const res = await requestWithSupertest.get(
+      '/api/tables/users/rows?_filters=createdAt__gte:2009-01-01 00:00:00'
+    );
+
```
Comment (IanMayo) [inline core/src/controllers/rows.test.js:76]: thanks @AbegaM - could we also do a more pessimistic test too, please?

So:
* ask for `gte` for a date in 2010
* a `lte` for a date in 2008
* plus `_eq` and `neq` for specific dates? (test `__eq` for a date that is present, and a date that isn't present)

**Review by IanMayo — DISMISSED** [review]
Tested and working

**Comment by thevahidal** [comment]
Thanks for the PR @AbegaM, can you please bump the version so we can merge it?

**Review by IanMayo — APPROVED** [review]

**Pull request merged**